comm.cpp

```diff
@@ -246,13 +244,24 @@ int Endpoint::read_msg(struct buffer *pbuf, int *target_sysid, int *target_compi
     _last_packet_len = expected_size;
     _read_total++;
 
-    if (_crc_check_enabled && !_check_crc())
-        return 0;
+    if (_crc_check_enabled) {
+        msg_entry = mavlink_get_msg_entry(msg_id);
```

Comment:
How, if `_crc_check()` is called after `mavlink_get_msg_entry()`? Indeed, result of later is argument for first...